### PR TITLE
Added support for custom Game Collection creation

### PR DESCRIPTION
**Pull request by pjft**
Summary of PR:

This is part 2 of the "Game Collections" effort. In particular, this allows users to create a new collection, either from an unused theme folder name (i.e. "neogeo", "capcom", whatever) or by specifying their own name. 

In terms of options:
- Users can choose to keep all custom collections at the top/system level (relying on jrassa's default theme changes), or bundle any collections for which no folder with the same name has been found under the "Collections" system, as sub-folders.
- Users can then choose to keep the normal sorting (current ES_systems ones, and then the new ones), or sort all systems together, in which case it'll sort them all alphabetically, and keep "RetroPie" and the 4 automatic ones at the end. 

Compared to the first PR, I refactored runtime collection creation to be on-demand rather than on startup, so that it only loads collections if they're being used, rather than loading them in memory but just not displaying them. This should hopefully reduce the impact of collections to be negligible in setups where they are not being used.

This has been extensively tested in the community for the past 25 days:

https://retropie.org.uk/forum/topic/11805/please-test-creating-custom-collections

Some user-created videos:
https://www.youtube.com/watch?v=eYG90Q6Smu0
https://www.youtube.com/watch?v=1PcTcldpAj4
https://www.youtube.com/watch?v=kkbhCf3CQaE (this one was the one I made and then noticed a bug during the recording, but can be used as well for an example)

I added some final fixes, code tweaks, and rebased it today, so bar any particular issues that crept up from the rebasing, it should still be working fine.

The code is reasonable, I tried to clean it up and document it quite a bit but after all these weeks stuck in this code I feel that I have kind of lost perspective of what can be tweaked and refactored anymore. I'm sure there's plenty there that I'm just no longer seeing, so if there's anything particularly critical I'd like to handle it.
As such, I'd appreciate the usual kindness of anyone who's willing to go through it and share any feedback on how the code could be better. 

If you want to test it functionally, I'm also more than happy to have it, but I believe that that should no longer be a bit issue as several users have been using it (and myself as well) regularly for the past 25 days.

Thanks in advance for your time, and apologies for the delay with this. Real life got in the way.

### CODE REVIEW

**Comment by TheyCallMeCoco** [comment]
I've tested this new feature almost from day one. Any issues I've seen have been resolved by pjft already. It's a great new addition to EmulationStation and can't wait to see it in the master branch.

**Comment by Darknior** [comment]
Same, it will be great to have it in master branch <3

**Comment by dmmarti** [comment]
I've been testing this using multiple themes and so far, all of the bugs I initially encountered have been dealt with.  The new feature is working great as pjft described above.  Awesome ES enhancement to RetroPie.

**Comment by jrassa** [comment]
@pjft great job.  Just gave it a final run through on my pi and everything works great.

**Comment by jrassa** [comment]
@pjft did just to confirm does this contain the fix for #185?

**Comment by pjft** [comment]
Oh, wow - thank you for the fast turnaround. If you or anyone do have
feedback on the code quality, definitely say so and I'll add it in a
separate PR, that was my main concern after being stuck in this code for
the past several weeks.

As for the fix, it certainly should have unless that's a different instance
of the bug. I needed to fix that as the Collections will rely on folders if
unthemed, so it became a lot more prevalent.

All in all, it was just a missing setCursor after opening the folder.

Thanks all.

File: es-app/src/views/gamelist/ISimpleGameListView.cpp [hunk 1]
```diff
@@ -93,6 +93,8 @@ bool ISimpleGameListView::input(InputConfig* config, Input input)
 				{
 					mCursorStack.push(cursor);
 					populateList(cursor->getChildrenListToDisplay());
+					FileData* cursor = getCursor();
+					setCursor(cursor);
```
Comment (pjft) [inline es-app/src/views/gamelist/ISimpleGameListView.cpp]: @zigurana - I think these two lines are what fixed the images/metadata not showing when opening the folder, if I recall correctly.

**Pull request merged**